### ISSUE TITLE
Uncaught TypeError: $scope.modelValue.splice is not a function

### ISSUE THREAD

**Issue by tino415** [a5hik/ng-sortable#197]
This I get on drop, I'm using controller as syntax.

My template:

````html
 <ul class="nav navbar-nav navbar-right" as-sortable="NCtrl.sortableFunctions" ng-model="NCtrl.pages">
    <li ng-repeat="page in NCtrl.pages"
            as-sortable-item
           data-match-route="edit/{{page.page.template}}/{{page.page.link}}"
        >
        <a as-sortable-item-handle href="#edit/{{page.page.template}}/{{page.page.link}}">
            {{page.page.title}}
        </a>
    </li>

    <!--
    <li ng-repeat-end class="navbar-btn" ng-show="NCtrl.editMode">
        <button class="btn btn-danger" ng-click="NCtrl.remove(page.id)">Remove</button>
    </li>
    -->
</ul>
````

Sortable functions:

```js
vm.sortableFunctions.orderChanged = function() { console.log('order change');}; // triggered when item order is changed with in the same column.
vm.sortableFunctions.itemMoved = function() { console.log('order change');}; // triggered when an item is moved accross columns.
vm.sortableFunctions.dragStart = function() { console.log('order change');};// triggered on drag start.
vm.sortableFunctions.dragEnd = function() { console.log('order change');};// triggered on drag end.
```

**Comment by snoblucha** [comment]
I got simmilar error, but most probably for a different reason. 

For me the reason was I used ng-model="options".  This is a pitfall. This scope value is overwritten later in the ng-sortable directive code to the options from as-sortable attribute. 

The cause of this error is either NCtrl.pages is not an array, or it is overwritten somewhere in the directive. I recommend to set breakpoint to exception in inspector and  see what is really in the `$scope.modelValue`.
